### PR TITLE
Forbid understanding the messages on the signs for illiterate characters

### PR DESCRIPTION
**Pull request by Night-Pryanik**

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Jenkins, rebuild

**Comment by AMurkin** [comment]
Why not like so?
```c++
    std::string signage = g->u.has_trait( trait_ILLITERATE ) ? _( "Sign: ???" ) : m.get_signage( lp );
    if( !signage.empty() && signage.size() < 36 ) {
         mvwprintw(w_look, ++line, column, _("Sign: %s"), signage.c_str());
    } else if( !signage.empty() ) {
        // Truncate to width of window as a guesstimate.
        mvwprintw(w_look, ++line, column, _("Sign: %s..."), signage.substr(0, 32).c_str());
    }
```

File: src/game.cpp [hunk 1]
```diff
@@ -7853,12 +7854,12 @@ void game::print_terrain_info( const tripoint &lp, WINDOW *w_look, int column, i
         wprintz(w_look, ll.second, ll.first.c_str());
     }
 
-    std::string signage = m.get_signage( lp );
-    if (!signage.empty() && signage.size() < 36) {
-        mvwprintw(w_look, ++line, column, _("Sign: %s"), signage.c_str());
-    } else if (!signage.empty()) {
+    std::string signage = g->u.has_trait( trait_ILLITERATE ) ? _( "???" ) : m.get_signage( lp );
```
Comment (BevapDin) [inline src/game.cpp]: Don't access `u` via the global `g` pointer, this code is already inside a `game` member function, use just `u`.

**Comment by Leland** [comment]
A nice touch.

I'd consider updating this line to include "signs" as well:

https://github.com/CleverRaven/Cataclysm-DDA/blob/e612c48f9ee01b60e87eab4336c0fa8498b52787/data/json/mutations.json#L606

**Pull request merged**